src/System.Management.Automation/engine/runtime/Binding/Binders.cs

```diff
@@ -7135,16 +7130,20 @@ internal static Expression InvokeMethod(MethodBase mi, DynamicMetaObject target,
 
                     if (expandParameters)
                     {
-                        argExprs[i] = Expression.NewArrayInit(
-                            paramElementType,
-                            args.Skip(i).Select(
-                                a => a.CastOrConvertMethodArgument(
+                        IEnumerable<Expression> elements = args
+                            .Skip(i)
+                            .Select(a =>
+                                a.CastOrConvertMethodArgument(
                                     paramElementType,
                                     paramName,
                                     mi.Name,
                                     allowCastingToByRefLikeType: false,
                                     temps,
-                                    initTemps)));
+                                    initTemps));
```

Comment:
So there won't be doubles of the actual code in the compiled result, but `CastOrConvertMethodArgument` will still run twice. It's probably worth doing `.ToArray()` here to avoid that